nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -83,24 +83,27 @@ def _setup(self, config):
             f"_setup: trial={self._trial_info.trial_name}({self.iteration}), "
             f"config={config}")
 
+        # Get checkpoint file to restore the training from
+        self.restore_checkpoint_file = config.pop("restore_checkpoint_file", None)
+
         # Try to recover a trial at least this many times
         self.max_retries = max(config.get("max_retries", 3), 0)
 
         # Create ray remote workers
         self._create_workers(config)
 
-        # Save initialized model
-        if config.get("checkpoint_at_init", False):
-            self.save()
-
         # Load initial state from checkpoint file
-        self._checkpoint_file = config.get("checkpoint_file", None)
-        if self._checkpoint_file is not None:
-            with open(self._checkpoint_file, mode="rb") as f:
+        self._restored = False
+        if self.restore_checkpoint_file is not None:
+            with open(self.restore_checkpoint_file, mode="rb") as f:
                 state = pickle.load(f)
                 self._restore(state)
                 self._restored = True
 
+        elif config.get("checkpoint_at_init", False):
```

Comment:
For consistency, wouldn't we still want to save the checkpoint at init if that flag is set to True?

Reply:
Are you sure? It would just be a copy of the original checkpoint?

Reply:
It would, just wondering for consistency. I'm fine either way.